### PR TITLE
Script for rotating monitor created. It takes one paramter. The direc…

### PR DESCRIPTION
**Pull request by dals83**
…tion of the rotation. The scripts is based on xrandr command.

### CODE REVIEW

File: admin_scripts/rotate_display.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+#-    version         rotate_display.sh (magenta.dk) 1.0.0
+#-    author          Danni Als
+#-    copyright       Copyright 2019, Magenta ApS
+#-    license         GNU General Public License
+#-    email           danni@magenta-aps.dk
+#-
+#================================================================
+#  HISTORY
+#     2019/11/12 : af : Script created
+#     2019/11/14 : af : Changed tactics. Instead of trying to connect to the X-server from root, we now rotate screen upon user login.
```
Comment (magenta-af) [inline admin_scripts/rotate_display.sh]: Your scripts give me far more credit than I deserve. :-)

File: admin_scripts/rotate_display.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    echo "Wrong rotation command given: $1"
+    exit -1
+fi
+
+AUTOSTART_FOLDER=/home/.skjult/.config/autostart/
+
+if [ ! -d $AUTOSTART_FOLDER ]; then
+    mkdir $AUTOSTART_FOLDER
+    echo "$AUTOSTART_FOLDER created..."
+fi
```
Comment (magenta-af) [inline admin_scripts/rotate_display.sh]: One simpler option here is the `mkdir -p` command, which also makes all the folders necessary to make the destination folder and which _doesn't_ fail if the folder already exists.

File: admin_scripts/rotate_display.sh [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+ROTATE_SCREEN_FILE=rotate_screen.sh
+ROTATE_SCREEN_FILE_COMPLETE_PATH="$AUTOSTART_FOLDER$ROTATE_SCREEN_FILE"
+
+if [ -f $ROTATE_SCREEN_FILE_COMPLETE_PATH ]; then
+    rm $ROTATE_SCREEN_FILE_COMPLETE_PATH
+    echo "$ROTATE_SCREEN_FILE_COMPLETE_PATH deleted..."
+fi
+cat <<EOT >> "$ROTATE_SCREEN_FILE_COMPLETE_PATH"
```
Comment (magenta-af) [inline admin_scripts/rotate_display.sh]: There are two simpler alternatives here: you can either do `rm -f "$ROTATE_SCREEN_FILE_COMPLETE_PATH"` to make sure the file's gone (in a way that doesn't fail if it doesn't exist), or you can just change the redirection operator to `>` to overwrite it.

File: admin_scripts/rotate_display.sh [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+DESKTOP_FILENAME=rotate_screen.desktop
+DESKTOPFILE_COMPLETE_PATH="$AUTOSTART_FOLDER$DESKTOP_FILENAME"
+if [ -f $DESKTOPFILE_COMPLETE_PATH ]; then
+  exit 0
+fi
+
+#xrandr --output DP-1 --rotate $1
+
+cat <<EOT >> "$DESKTOPFILE_COMPLETE_PATH"
```
Comment (magenta-af) [inline admin_scripts/rotate_display.sh]: This _could_ also be simplified, but I suppose it's less important if you want to skip creating this file.

File: admin_scripts/rotate_display.sh [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+Hidden=false
+NoDisplay=false
+X-GNOME-Autostart-enabled=true
+Name[en_US]=rotate screen
+Name=rotate screen
+Comment[en_US]=Rotates Primary monitor $1
+Comment=Rotates Primary monitor $1
+EOT
+
+chmod +x $DESKTOPFILE_COMPLETE_PATH
```
Comment (magenta-af) [inline admin_scripts/rotate_display.sh]: (As an aside, I didn't know this is necessary, but you're quite right: the [Ubuntu security policy](https://wiki.ubuntu.com/SecurityTeam/Policies#Execute-Permission_Bit_Required) requires that desktop files in the home folder have the execute bit set.)

File: admin_scripts/rotate_display.sh (hunk 2 above)
Comment (dals83) [reply admin_scripts/rotate_display.sh]: Yes but then we cant see if it was created or not unless the command gives us output of what happened.

File: admin_scripts/rotate_display.sh (hunk 3 above)
Comment (dals83) [reply admin_scripts/rotate_display.sh]: Again I think it is important that we can see in the log what precisely happened so we know which state the filesystem is in...

Comment (magenta-af) [reply admin_scripts/rotate_display.sh]: In that case, why not change the `set -e` at the top to `set -xe` to print out every command as it's run? Then the log gives a (more or less) complete view of everything that happened.

**Pull request merged**